syntax = "proto3";

service Messenger {
    rpc GetDump (Date) returns (Dump) {}
    rpc HasUpdate (Date) returns (UpdateStatus) {}  
}

message Location {
    float latitude = 1;
    float longitude = 2;
}

message Date {
    int32 month = 1;
    int32 day = 2;
    int32 year = 3;
}

message Row {
    Location location = 1;
    Date date = 2;
    int32 confirmed_cases = 3;
    int32 confirmed_deaths = 4;
    int32 confirmed_recovered = 5;
}

message Dump {
    repeated Row rows = 1;
}

message UpdateStatus {
    bool pull = 1;
}
